Add nvm-exec to the nvm formula.

hjdivad opened the pull request:
`nvm exec VERSION SCRIPT` depends on `nvm-exec` existing in `$NVM_DIR`.  The current formula does not download `nvm-exec` which results in errors when users try to `nvm exec`.

hjdivad commented:
I've also updated the caveats to separate the bits that are wholly optional from the bit that is actually important.

dunn commented on Library/Formula/nvm.rb:
```diff
@@ -4,21 +4,42 @@ class Nvm < Formula
     bash_completion.install "bash_completion" => "nvm"
+
+    resource("nvm-exec").stage do
+      prefix.install("nvm-exec")
```
should this be `bin.install`?

hjdivad commented:
@jacknagel @mikemcquaid I think I've followed the contributing guidelines for this, but if there's anything you need me to change to help the PR along just let me know.  Thanks! ^_^

mikemcquaid commented on Library/Formula/nvm.rb:
```diff
@@ -4,21 +4,42 @@ class Nvm < Formula
+
+        source $(brew --prefix nvm)/nvm.sh
+
+
```
Can remove at least one (if not all) the empty lines here.

hjdivad replied:
will do

mikemcquaid commented on Library/Formula/nvm.rb:
```diff
@@ -4,21 +4,42 @@ class Nvm < Formula
         export NVM_DIR=~/.nvm
-        source $(brew --prefix nvm)/nvm.sh
 
       Type `nvm help` for further information.
```
Can we slim these caveats a lot? There's a lot of text here and it feels like it's not all Homebrew-specific (only Homebrew-specific bits should be here).

hjdivad replied:
Yeah.  After doing [some digging](https://github.com/Homebrew/homebrew/commit/8879b432be43507f2178cfe1628edc5f48554d42#diff-1f962c21e02f5c6f88fb1a6e323538a1) I see the point of putting nvm in the user's home directory is to avoid losing node installs on upgrade. 

It seems unfortunate because it adds some steps to the install process but oh well :-/  I'll revert the instructions and only add the line about copying `nvm-exec`.

mikemcquaid commented on Library/Formula/nvm.rb:
```diff
@@ -4,21 +4,42 @@ class Nvm < Formula
-      equivalent configuration file:
+      Copy nvm-exec to NVM's working directory
+
+        cp $(brew --prefix nvm)/nvm-exec ~/.nvm/
```
If you have to manually copy this I wonder if it's valuable to install it at all?

hjdivad replied:
As of right now, if you `brew install nvm` you will have a version of `nvm` that is not able to `nvm exec VERSION SCRIPT`.  Trying to do so will get you a cryptic error.

So for a fully functioning `nvm` the user needs to have this script and it needs to wind up in their `NVM_DIR`.

Either homebrew downloads and checks the script or the caveats section should have a line that does that.  I think the value in having homebrew do the checking is the resulting instructions for the user are easier to understand (copy this file vs download & check this file) although I'll admit not by much.

@mikemcquaid  would you prefer the formula to just tell the user how to download and check the script?

hjdivad commented:
@mikemcquaid i've updated the caveats section to describe setting up `~/.nvm` as required, the way it was before and added the line to copy `nvm-exec`.

As far as I know the only thing we need to do is figure out whether we prefer to:
1. download and check the `nvm-exec` script, and instruct the user to copy it to their `NVM_DIR`.
2. update caveats to show the user how to download and check the `nvm-exec` script.

What's your preference?

mikemcquaid commented:
@hjdivad This looks good now, thanks!

mikemcquaid commented:
Thanks for your contribution to Homebrew! Without people like you submitting PRs we couldn't run this project. You rock!

For future reference the preferred commit message format is `nvm: add nvm-exec.`.

hjdivad commented:
@mikemcquaid ah thanks for the heads up.  I'll try to keep the preferred commit message format in mind for the next PR.  And thanks for maintaining homebrew.  As I'm sure you know, it's a lifesaver ^_^
